waspc/data/Generator/templates/server/src/auth/providers/email/login.ts

```diff
@@ -12,25 +17,27 @@ export function getLoginRoute({
         req: Request<{ email: string; password: string; }>,
         res: Response,
     ): Promise<Response<{ token: string } | undefined>> {
-        const userFields = req.body || {}
-        ensureValidArgs(userFields)
+        const fields = req.body || {}
+        ensureValidArgs(fields)
 
-        userFields.email = userFields.email.toLowerCase()
+        fields.email = fields.email.toLowerCase()
 
-        const user = await findUserBy({ email: userFields.email })
-        if (!user) {
+        const authIdentity = await findAuthIdentity("email", fields.email)
+        if (!authIdentity) {
             throwInvalidCredentialsError()
         }
-        if (!user.isEmailVerified && !allowUnverifiedLogin) {
+        const providerData = deserializeProviderData(authIdentity.providerData)
```

Comment:
Everywhere where we use `providerData` we are manually doing the serialisation and deserialisation process:
1. Parse / stringify JSON (we cannot use JSON field but a String field due to [lack of Prisma's support for SQLite JSON field](https://github.com/prisma/prisma/issues/3786))
2. Hash the password